### PR TITLE
[RLlib] Examples folder cleanup: ModelV2 -> RLModule wrapper for migrating to new API stack (by config).

### PR DESCRIPTION
**Pull request by sven1977**
Adds example script and RLModule example class for easy ModelV2 -> RLModule migration to new API stack using an old API stack config.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/examples/rl_modules/classes/modelv2_to_rlm.py [hunk 1]
```diff
@@ -52,7 +119,31 @@ def setup(self):
...
+        if state_in and seq_lens is None:
+            seq_lens = torch.tensor(
+                [1.0] * state_in[0].shape[0], device=state_in[0].device
+            )
+        # Perform the actual ModelV2 forward pass.
+        # A recurrent ModelV2 adds and removes the time-rank itself (whereas in the new
+        # API stack, the connector pipelines are responsible for doing this) -> We have
+        # to remove, then re-add the time rank here to make ModelV2 work with the new
+        # model-to-env pipelines. 
+        if state_in:
```
Comment (simonsays1980) [inline rllib/examples/rl_modules/classes/modelv2_to_rlm.py:138]: Ah yes, now LSTM is possible :)

File: rllib/examples/rl_modules/migrate_modelv2_to_new_api_stack_by_config.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        .api_stack(
+            enable_rl_module_and_learner=True,
+            enable_env_runner_and_connector_v2=True,
+        )
+        .rl_module(
+            rl_module_spec=RLModuleSpec(
+                module_class=ModelV2ToRLModule,
+                model_config_dict={
+                    "policy_id": DEFAULT_POLICY_ID,
+                    "algo_config": config_old_stack,
```
Comment (simonsays1980) [inline rllib/examples/rl_modules/migrate_modelv2_to_new_api_stack_by_config.py:44]: Can we call this specifically `"old_api_stack_algo_config"` to make clear that this is something only for this kind of purpose?

**Review by simonsays1980 — APPROVED** [review]
LGTM.

File: rllib/examples/rl_modules/migrate_modelv2_to_new_api_stack_by_config.py (hunk 2 above)
Comment (sven1977) [reply rllib/examples/rl_modules/migrate_modelv2_to_new_api_stack_by_config.py:44]: good point! Will change ...

Comment (sven1977) [reply rllib/examples/rl_modules/migrate_modelv2_to_new_api_stack_by_config.py]: done

**Pull request merged**